C++/test/algorithm/sorting/sorting.cpp

```diff
@@ -10,20 +10,23 @@
 #include "algorithm/sorting/quick_sort.hpp"
 #include "algorithm/sorting/selection_sort.hpp"
 #include "algorithm/sorting/shell_sort.hpp"
+#include "algorithm/sorting/bucket_sort.hpp"
 
 // Prototypes
 int generate_random_int(int, int);
 vector<int> generate_unsorted_vector(int max_size = 1000);
 
 // Pointer to function
-using sorting_function = void(*)(vector<int>&, int, bool);
+using SortingFunction = void(*)(vector<int>&, int, bool);
+
+SortingFunction bucket_insertion = bucket_sort<insertion_sort>;
```

Comment:
Had a chance to look through this today. Obviously your implementation works and the code looks good, but I'd rather we didn't use our internal insertion sort method in `sorting.cpp`. A couple reasons:

Consistency: I'd like to keep `sorting.cpp` as algorithm-agnostic as possible. In other words, I'd rather make the sorting algorithm header files fit the sorting test rather than modifying the sorting test each time we add a new algorithm. Modifying the sorting test to fit the sorting algorithms (as you do with the template and `insertion_sort`) defeats the purpose of having a common sorting algorithm.
 
Clarity: The algorithm is easier to track if you don't have to move to another file to figure out how the sorting happens. Since we're generally oriented toward people who are newer to programming, I think that this is important.

With this in mind, here are some suggestions of other ways we could do this: 

- Implement the sorting manually in `bucket_sort.hpp`.
- Use the sorting function from the STL (since demonstrating this algorithm is more about showing how the buckets work and less about showing how each individual bucket is sorted).
- Include `insertion_sort.hpp` directly into `bucket_sort.hpp`, though I think this is my least favorite option as I'd rather that the algorithms were not dependent on one another (modifying `insertion_sort.hpp` at some point in the future shouldn't break `bucket_sort.hpp`).

OK, I've already written way too much about this. Please don't hesitate to push back if you feel like I'm missing the point of your code if you think your way is still better!

Reply:
Hi @alxmjo, nice to meet :)
Sorry for the delay.

I fixed the README conflicts and went with your second suggestion about the internal sort. I kept the templated implementation because I think its it actually helps with understanding the essence of this algorithm but I used a default parameter and provided a wrapper for the STL implementation, compatible with this repo's sorting function interface. I also minimized the impact on sorting.cpp so that bucket sort doesn't need "special treatment".

Hope this is OK.

Thanks!